packages/front-end/components/AuthorizedImage.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { useEffect, useState, FC } from "react";
+import { FaExclamationTriangle } from "react-icons/fa";
+import { useAuth } from "@/services/auth";
+import { getApiHost } from "@/services/env";
+import LoadingSpinner from "./LoadingSpinner";
+
+interface AuthorizedImageProps extends React.HTMLProps<HTMLImageElement> {
+  imageCache?: Record<string, string>;
+}
+
+/**
+ * This component is used to display images that are stored in a private bucket.
+ * It will fetch the image from the backend and convert it to a blob url that can be displayed.
+ * It will also cache the image if imageCache is set so that it does not need to be fetched again.
+ * */
+const AuthorizedImage: FC<AuthorizedImageProps> = ({
+  imageCache = {},
+  src = "",
+  ...props
+}) => {
+  const [imageSrc, setImageSrc] = useState<string | null>(null);
+  const [error, setError] = useState<string | null>(null);
+
+  const { apiCall } = useAuth();
+
+  useEffect(() => {
+    const fetchData = async (url) => {
+      try {
+        const imageData: Blob = await apiCall(new URL(url).pathname);
+        const imageUrl = URL.createObjectURL(imageData);
+        imageCache[url] = imageUrl;
+        setImageSrc(imageUrl);
+      } catch (error) {
+        setError(error.message);
+      }
+    };
+
+    const s3pattern = /^https:\/\/([a-z0-9-]+)\.s3\.amazonaws\.com\/(.*)$/;
```

Comment:
> The frontend doesn't have access to backend env. So I couldn't check to see s3 bucket name. So any link to any s3 bucket will get routed through us, even if someone in Markdown links to a different s3 bucket. But the likelihood of that is small - and if they do it would have to be a public bucket anyway and so the endpoint should work.